src/tests/multihost/ipa/test_subid_ranges.py

```diff
@@ -0,0 +1,132 @@
+""" Automation of IPA bugs """
+
+import pytest
+import subprocess
+import time
+import os
+from sssd.testlib.common.utils import SSHClient
+
+
+test_password = "Secret123"
+user = 'admin'
+
+
+def execute_cmd(multihost, command):
+    """ Execute command on client """
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+@pytest.mark.usefixtures('install_list_subid_ranges',
+                         'setup_ipa_client')
+@pytest.mark.tier1
+class TestSubid(object):
+    """
+    This is for ipa bugs automation
+    """
+    def test_subid_feature(self, multihost,
+                           create_bkp_for_subid_files):
+        """
+        :Title: support subid ranges managed by FreeIPA
+        :id: 50bcdc28-00c8-11ec-bef4-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1803943
+        :steps:
+            1. Generate subid for user admin
+            2. Test newuidmap command
+            3. Test newgidmap command
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        ssh1 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password=test_password)
+        (result, result1, exit_status) = ssh1.execute_cmd('kinit',
+                                                          stdin=test_password)
+        assert exit_status == 0
+        (result, result1, exit_status) = ssh1.exec_command('klist')
+        assert user in str(result1.read())
+        (result, result1, exit_status) = ssh1.exec_command(f"ipa "
+                                                           f"subid-generate "
+                                                           f"--owner={user}")
+        (result, result1, exit_status) = ssh1.exec_command(f"ipa "
+                                                           f"subid-find "
+                                                           f"--owner "
+                                                           f"{user}")
+        user_details = result1.readlines()
+        uid_start = int(user_details[5].split(': ')[1].split('\n')[0])
+        uid_range = int(user_details[6].split(': ')[1].split('\n')[0])
+        gid_start = int(user_details[7].split(': ')[1].split('\n')[0])
+        gid_range = int(user_details[8].split(': ')[1].split('\n')[0])
+        ssh2 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password=test_password)
+        cmd = execute_cmd(multihost, "ps -ef | grep bash").stdout_text
+        find_admin = f"ps -ef | grep bash | grep {user}"
```

Comment:
Why is this two steps: `ps -ef | grep bash` + `ps -ef | grep bash | grep {user}`?
Wouldn't it be possible to do only second `grep`?

Reply:
Its to maintain word limit in python for gitgub its 79 word per line . If i put all in one single line it will cross 79 words . That is why i split ed the line in two

Reply:
My question is - why do you need to execute two commands:
 - "ps -ef | grep bash"
 - "ps -ef | grep bash | grep {user}"

Reply:
cmd = execute_cmd(multihost, "ps -ef | grep bash").stdout_text  ---  This one is used to find any preexisting admin process  
find_admin = f"ps -ef | grep bash | grep {user}" ---  This one is used to find current process id of user admin 

But i think one one can be used . I will fix it

Reply:
Right, I think both things can be done via single `ps`.
But then the question is, why is it required to terminated those shells at all.

Reply:
There might be some other process running for admin from previous test , we dont know when this test script will be executed and after which test . So for this test to work correctly , here i am making sure that no other admin process are there by killing any other admin user process

Reply:
cmd = execute_cmd(multihost, "ps -ef | grep bash").stdout_text  ---  this command is directly executed on client machine.
        find_admin = f"ps -ef | grep bash | grep {user}" ---  this is just a variable , i need this to reduce word limit to 79 

So i need both the line

Reply:
Variable isn't what I'm talking about.
Two `execute_cmd('ps')` is.

Reply:
Now there is only "ps"

Reply:
No, you still execute `ps` twice. And I don't understand a reason for this.
You can compose a `proces_id` list via single `ps`.

Reply:
Now only one ps

Reply:
Expectation that element [0] always contains proper `pid` is wrong.

There were 2 x `ps` to terminate admin shells and 1 x `ps` to find a pid.

I asked to:
1) either find a way to get `pid` without `ps` at all (something like `echo $$ > /tmp/pid`)
2) or remove one of first 2 `ps` (not both altogether with termination)

Reply:
fixed